src/app/App.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { Component } from "react";
+import HomeContainer from "./home/HomeContainer";
+import LoginContainer from "./auth/LoginContainer";
+import { Router, Route, Switch, Redirect } from "react-router-dom";
+import PrivateRoute from "./auth/PrivateRoute";
+import { connect } from "react-redux";
+import { push } from "connected-react-router";
+
+class AppComponent extends Component<any, any> {
```

Comment:
> And the last question, why to change folder structure? Because in case of redux there will be a lot of shared reducers and using one folder and one file for reducers and actions this is not very good idea, as the result it will grow, so it is better to use global reducers folder, same as actions folders.

As a team we decided on using a feature based approach for our directory structure. This allows for us to have a directory for each part of our app (Home page, login/auth, etc) that will have its own reducers, operations, actions, types, etc. 
https://github.com/erikras/ducks-modular-redux
https://medium.freecodecamp.org/scaling-your-redux-app-with-ducks-6115955638be